### PR TITLE
Category controller

### PR DESCRIPTION
**Pull request by TuenFV**
Category controller only

### CODE REVIEW

File: app/controllers/admin/categories_controller.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    end
+    
+  end
+
+  def edit
+  end
+
+  def update
+    @category.update(params_category)
+    if @category.update(params_category)
```
Comment (duyetvn) [inline app/controllers/admin/categories_controller.rb]: duplicate

File: app/controllers/admin/categories_controller.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      flash[:error] = "Can not update!"
+      render :edit
+    end
+  end
+
+  def show
+  end
+
+  def destroy
+    @category.destroy
```
Comment (duyetvn) [inline app/controllers/admin/categories_controller.rb:43]: flash

File: app/views/admin/categories/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<%= form_with(model: [ :admin, category], local: true) do |f| %>
+
+  <% if category.errors.any? %>
+    <% category.errors.each do |error| %>
+      <li><%= error.full_message %></li>
+    <% end %>
+  <% end %>
```
Comment (duyetvn) [inline app/views/admin/categories/_form.html.erb]: separate to _message.html.erb

File: app/views/admin/categories/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<div>
+	<h1>Hello from SHOW action</h1>
+</div>
+
+<%= form_with(model: [:admin , @category]) do |f| %>
```
Comment (duyetvn) [inline app/views/admin/categories/show.html.erb]: Show thì cần gì form?

File: app/controllers/admin/categories_controller.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+  def create
+    @category = Category.new(params_category)
+    @category.save
+    
+    if @category.save
+      flash[:notice] = 'Success!'
+      redirect_to admin_categories_path
+    else
+      flash[] = 'Error!'
```
Comment (duyetvn) [inline app/controllers/admin/categories_controller.rb]: flash.now[:error] = @category.errors.full_messages

File: app/controllers/admin/categories_controller.rb [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  def edit
+  end
+
+  def update
+    @category.update(params_category)
+    if @category.update(params_category)
+      flash[:notice] = "Sucess update!"
+      redirect_to admin_category_path
+    else
+      flash[:error] = "Can not update!"
```
Comment (duyetvn) [inline app/controllers/admin/categories_controller.rb]: same create

File: app/views/admin/categories/_message.html.erb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+<% if @category.errors.any? %>
+  <% @category.errors.each do |error| %>
+    <li><%= error.full_message %></li>
+  <% end %>
+<% end %>
```
Comment (duyetvn) [inline app/views/admin/categories/_message.html.erb]: Use flash

<% flash.each do |key, value| %>
....
<% end %>

File: app/views/admin/categories/_form.html.erb [hunk 8]
```diff
@@ -1,10 +1,6 @@
 <%= form_with(model: [ :admin, category], local: true) do |f| %>
 
-  <% if category.errors.any? %>
-    <% category.errors.each do |error| %>
-      <li><%= error.full_message %></li>
-    <% end %>
-  <% end %>
+  <%= render 'message' %>
```
Comment (duyetvn) [inline app/views/admin/categories/_form.html.erb]: render on application.html.erb

File: app/views/admin/categories/_message.html.erb (hunk 7 above)
Comment (duyetvn) [reply app/views/admin/categories/_message.html.erb]: move to layout folder

File: app/controllers/admin/categories_controller.rb [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+  def edit
+  end
+
+  def update
+    if @category.update(params_category)
+      flash[:notice] = "Category was sucessfully updated!"
+      redirect_to admin_category_path
+    else
+      flash[:alert] = @category.errors.full_messages
+      redirect_to edit_admin_category_path
```
Comment (duyetvn) [inline app/controllers/admin/categories_controller.rb:35]: render :edit, status: 422

**Comment by TuenFV** [comment]
yes sir

**Pull request merged**